### PR TITLE
Fix possible crash on showing message actions

### PR DESCRIPTION
**Pull request by cardoso**
Came across this crash while playing around with the example & tapping messages randomly. I couldn't reproduce a second time, but haven't tried much.

Screen Shot 2020-03-30 at 8 44 01 PM

### CODE REVIEW

**Comment by b-onc** [comment]
Hello @cardoso ,
Thanks for finding and fixing this bug!
We'd like to have a general solution for this OOB problem, so we have this solution in mind:
Inside `Sources/Client/Extensions`
```swift
extension Array {
    func element(at index: Array.Index) -> Element? {
        guard index < count, index >= 0 else { return nil }
        return self[index]
    }
}
```
so we can easily:
```swift
guard let elem = array.element(at: index) else { return }
```
and use the extension.
Also, there's no need to create the `UIContextMenuConfiguration` if there's no message for it.

It'd be perfect if you could make the changes before the end of today, since we're on a tight schedule and we'd like this bug fixed.

Again, thank you!

**Comment by cardoso** [comment]
Thanks @b-onc . I'll make the change

**Comment by cardoso** [comment]
@b-onc there's also plenty of places without OOB checking. Should I open a PR after this to use the safe subscript where applicable? Asking because some projects prefer crashes over having silent failures (I prefer apps that don't crash 🤷‍♂). As a compromise, I can add a log to the subscript so OOB issues can be visible during development.

File: Sources/UI/Chat/Chat View Controller/ChatViewController+MessageActions.swift [hunk 1]
```diff
@@ -311,11 +311,14 @@ extension ChatViewController {
             return nil
         }
         
+        guard let cell = tableView.cellForRow(at: indexPath),
+            let message = self.presenter?.items[safe: indexPath.row]?.message else {
```
Comment (b-onc) [inline Sources/UI/Chat/Chat View Controller/ChatViewController+MessageActions.swift:315]: Can you make sure this line aligns with the line above?

Comment (cardoso) [reply Sources/UI/Chat/Chat View Controller/ChatViewController+MessageActions.swift]: 👍

**Comment by b-onc** [comment]
> @b-onc there's also plenty of places without OOB checking. Should I open a PR after this to use the safe subscript where applicable? Asking because some projects prefer crashes over having silent failures (I prefer apps that don't crash 🤷‍♂). As a compromise, I can add a log to the subscript so OOB issues can be visible during development.

Thanks for the suggestion, currently it's not needed to act on every subscript. We also prefer to have a SDK without crashes, so we subscript whenever we don't expect a crash :)
Having patched this issue, it seems like we need to get down to the actual issue: why this function is being called while there's no message for this indexPath? We'll add this to our backlog and investigate. If you have any info on how you managed to produce the crash, please let us know!

**Comment by cardoso** [comment]
@b-onc got it 👍 . I had the crash while scrolling and tapping a few messages. It might have to do with pagination and/or some async call. I'll keep an eye out for it while I continue playing with the project.

**Comment by b-onc** [comment]
Last thing: the project currently doesn't build since the new file isn't added to the project. Can you add your new file to `StreamChatClient` target?

**Comment by b-onc** [comment]
@cardoso Is there anything I can do to help you get this merged?

**Comment by cardoso** [comment]
@b-onc I added the the new file to the project in the previous two commits. Am I still missing something?

**Comment by b-onc** [comment]
@cardoso There's a merge conflict. 
(FYI We squash/rebase merges)

**Comment by cardoso** [comment]
@b-onc you mean I should also fix this in master?
Screen Shot 2020-04-01 at 1 46 43 PM

**Comment by b-onc** [comment]
We use "Rebase and merge" option which is not possible currently - I see that there's merge conflicts. Interesting that it's all green for you, though.
image

**Comment by cardoso** [comment]
@b-onc I rebased via the CLI & pushed. It's all green here, so not sure what the effect is on your side.

**Comment by b-onc** [comment]
Seems that the issue is fixed with the push 15 minutes ago. Thanks!

**Comment by buh** [comment]
Great job @cardoso! Thanks!

**Pull request merged**